app/frontend/src/cljs/teet/project/project_controller.cljs

```diff
@@ -406,25 +418,40 @@
 
   UpdateBasicInformationForm
   (process-event [{:keys [form-data]} app]
-    (if (get-in app [:route :project :basic-information-form :road-info])
-      (update-in app [:route :project :basic-information-form]
-                 merge form-data)
-      (t/fx app
-            (fn [e!]
-              (e! (->FetchRoadInformation form-data))))))
-
-  FetchRoadInformation
-  (process-event [{form-data :form-data} app]
-    (let [{:thk.project/keys [road-nr carriageway]} (get-in app [:route :project])]
-      (t/fx (update-in app [:route :project :basic-information-form]
-                       merge form-data)
-            {:tuck.effect/type :rpc
-             :endpoint (get-in app [:config :api-url])
-             :method :GET
-             :rpc "road_info"
-             :args {:road road-nr
-                    :carriageway carriageway}
-             :result-path [:route :project :basic-information-form :road-info]})))
+    (log/info "UpdateBasicInformationForm" form-data)
+    (let [project-meter-range (fn [app]
+                                (let [{[start-km-string end-km-string] :thk.project/km-range}
+                                      (get-in app [:route :project :basic-information-form])]
+                                  [(or (some-> start-km-string road-model/parse-km road-model/km->m)
+                                       (get-in app [:route :project :thk.project/start-m]))
+                                   (or (some-> end-km-string road-model/parse-km road-model/km->m)
+                                       (get-in app [:route :project :thk.project/end-m]))]))
```

Comment:
Not related specifically to this PR, more of a general observation: Getting the meter range from the app state has become pretty hideous.